orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql

```diff
@@ -58,6 +58,21 @@ with
             count(distinct offer_id) as total_headlines
         from {{ ref("int_applicative__headline_offer") }}
         group by offer_id, is_headlined
+    ),
+
+    headline_grouped_by_offer as (
+        select
+            offer_id,
+            sum(total_headlines) as total_headlines,
+            max(is_headlined) as is_headlined,
+            min(first_headline_date) as first_headline_date,
+            case
+                when count(offer_id) > count(last_headline_date)
+                then null
+                else max(last_headline_date)
+            end as last_headline_date
+        from headline_offer_flattened
+        group by offer_id
     )
```

Comment:
je propose de faire un rank sur la last_endline_date pour que ça soit plus compréhensible par tous 
```suggestion
    with headline_offer_flattened as (
        select
            offer_id,
            case
                when
                    date(headline_ending_time) is null
                    or date(headline_ending_time) >= current_date
                then true
                else false
            end as is_headlined,
            min(date(headline_beginning_time)) as first_headline_date,
            max(date(headline_ending_time)) as last_headline_date,
            count(distinct offer_id) as total_headlines, 
        from `passculture-data-prod.int_applicative_prod.headline_offer`
        group by offer_id, is_headlined
), 
headline_offer_ranked as (
  select *, row_number() over(partition by offer_id order by last_headline_date) as rank_headline
  from headline_offer_flattened
  )
select offer_id,
total_headlines,
is_headlined, 
first_headline_date, 
last_headline_date
from headline_offer_ranked
where headline_offer_ranked is null
or rank_headline = 1
```